src/qibo/quantum_info/metrics.py

```diff
@@ -385,3 +385,101 @@ def gate_error(channel, target=None, backend=None):
     error = 1 - average_gate_fidelity(channel, target, backend=backend)
 
     return error
+
+
+def meyer_wallach_entanglement(circuit, backend=None):
+    """Computes the Meyer-Wallach entanglement Q of the `circuit`,
+
+    .. math::
+        Q = 1-\\frac{1}{N}\\sum_{k}\\text{Tr}\\left(\\rho_k^2(\\theta_i)\\right) \\,
+
+    Args:
+        circuit (:class:`qibo.models.Circuit`): Parametrized circuit.
+        backend (:class:`qibo.backends.abstract.Backend`, optional): backend to be used
+            in the execution. If ``None``, it uses ``GlobalBackend()``.
+            Defaults to ``None``.
+
+    Returns:
+        float: Meyer-Wallach entanglement.
+    """
+
+    if backend is None:  # pragma: no cover
+        backend = GlobalBackend()
+
+    circuit.density_matrix = True
+    nqubits = circuit.nqubits
+    rho = backend.execute_circuit(circuit).state()
+    entropy = 0
+    for j in range(nqubits):
+        trace_q = list(range(nqubits))
+        trace_q.pop(j)
+        rho_r = backend.partial_trace_density_matrix(rho, trace_q, nqubits)
+        trace = np.trace(rho_r @ rho_r)
+        entropy += trace
+
+    return 1 - entropy / nqubits
+
+
+def entangling_capability(circuit, samples: int, backend=None):
+    """Computes the average Meyer-Wallach entanglement Q of the `circuit`,
+
+    .. math::
+        Ent = \\frac{2}{S}\\sum_{k}Q_k \\, ,
+
+    where :math:`S` is the number of samples.
+
+    Args:
+        circuit (:class:`qibo.models.Circuit`): Parametrized circuit.
+        samples (int): number of samples to estimate the integral.
+        backend (:class:`qibo.backends.abstract.Backend`, optional): backend to be used
+            in the execution. If ``None``, it uses ``GlobalBackend()``.
+            Defaults to ``None``.
+
+    Returns:
+        float: Entangling capability.
+    """
+
+    if backend is None:  # pragma: no cover
+        backend = GlobalBackend()
+
+    res = []
+    nparams = 0
+    for gate in circuit.queue:
+        if hasattr(gate, "trainable") and gate.trainable:
+            nparams += len(gate.parameters)
+    for i in range(samples):
+        params = np.random.uniform(-np.pi, np.pi, nparams)
+        circuit.set_parameters(params)
```

Comment:
`nparams` value is already stored in a circuit as `circuit.trainable_gates.nparams`, so I the there is no need to go over the whole `circuit.queue`. Also, the Identity gate is considered `trainable` in Qibo so, if I understand correctly, it is better to avoid changing its parameters. The solution I would propose is:
```python
params = {
    gate: np.random.uniform(-np.pi, np.pi, gate.nparams)
    for gate in circuit.trainable_gates
    if isinstance(gate, gates.I) is False
}
circuit.set_parameters(params)
```

Reply:
Thanks @vodovozovaliza, this is much better. I will make the change when PR #895 is merged (also thank you for that).